bin/propolis-server/src/lib/stats/mod.rs

```diff
@@ -78,15 +82,80 @@ struct ServerStatsInner {
     /// data.
     virtual_machine: VirtualMachine,
 
+    /// A weak reference to the objects implementing the VM this Propolis server
+    /// is responsible for. It is critical this is a weak reference: we do not
+    /// want to prohibit `VmObjects` from being dropped when unneeded. The
+    /// referece is, however, necessary to measure "Propolis overhead" -
+    /// measurements of the Propolis process minus the VM it is managing.
+    vm_objects: Weak<VmObjects>,
+
     /// The reset count for the relevant instance.
     run_count: virtual_machine::Reset,
+
+    /// Mapped virtual memory for the Propolis managing this instance, excluding
+    /// VM memory.
+    vmm_vsz: virtual_machine::VmmVsz,
+
+    /// Process RSS is sampled in a standalone task. The `tokio::sync::watch`
+    /// here is to observe that sampling and update the `vmm_vsz` tracked
+    /// here.
+    process_stats_rx: tokio::sync::watch::Receiver<process::ProcessStats>,
```

Comment:
I'm not convinced that this actually needs to be a `tokio::sync::watch` --- a `watch` channel is basically just an `Arc`ed read-write lock plus a thing for notifying the readers when the value inside the `RwLock` changes, and we never use `watch::Receiver::changed` in this code, just `borrow()`. So, it's an `Arc<RwLock<ProcessStats>>` with extra overhead for functionality we're not using here...and, because there's only a single reader and a single writer, it may as well be a `Mutex` instead of a `RwLock`.

I would also maybe consider just stuffing `vsz` in an `AtomicUsize`, but Patrick will probably disagree with me...